bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyEngineWrapper.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jrubyscripting.internal;
+
+import java.io.Reader;
+
+import javax.script.Bindings;
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.Invocable;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineFactory;
+import javax.script.ScriptException;
+
+import org.jruby.embed.jsr223.JRubyEngine;
+
+/**
+ * This is a wrapper for {@link JRubyEngine}.
+ * 
+ * The purpose of this class is to intercept the call to eval and save the context into
+ * a global variable for use in the helper library.
+ *
+ * @author Jimmy Tanagra - Initial contribution
+ */
+public class JRubyEngineWrapper implements Compilable, Invocable, ScriptEngine {
+
+    private final JRubyEngine engine;
+
+    private final String CONTEXT_VAR_NAME = "ctx";
+    private final String GLOBAL_VAR_NAME = "$" + CONTEXT_VAR_NAME;
```

Comment:
This also causes some checkstyle warnings:
```suggestion
    private static final String CONTEXT_VAR_NAME = "ctx";
    private static final String GLOBAL_VAR_NAME = "$" + CONTEXT_VAR_NAME;
```